File: mergeSort.js
function mergeSort(arr) {
    let len = arr.length

    if (len > 1) {
        let mid = Math.round(len / 2);
        let left = arr.slice(0, mid)
        let right = arr.slice(mid)

        mergeSort(left)
        mergeSort(right)

        let i = 0
        let j = 0
        let k = 0
        
        while (i < left.length && j < right.length) {
            if (left[i] < right[j])
                arr[k++] = left[i++]
            else
                arr[k++] = right[j++]
        }
        while (i < left.length) 
            arr[k++] = left[i++]
        while (j < right.lenght)
            arr[k++] = right[j++]
    }
}

function main() {
    let arr = process.argv.slice(2)
    arr = arr.map(num => parseInt(num))
    arr = arr.filter(num => Number.isInteger(num))
    mergeSort(arr)
    console.log(arr)
}

if (require.main.children.length === 0)
    main()

File: mergeSort.py
import sys

def mergeSort(arr):
    n = len(arr)
    if n > 1:
        mid = n // 2
        left = arr[:mid]
        right = arr[mid:]
        mergeSort(left)
        mergeSort(right)

        i = 0
        j = 0
        k = 0
        while i < len(left) and j < len(right):
            if left[i] < right[j]:
                arr[k] = left[i]
                i += 1
            else:
                arr[k] = right[j]
                j += 1
            k += 1
        
        while i < len(left):
            arr[k] = left[i]
            i += 1
            k += 1
        while j < len(right):
            arr[k] = right[j]
            j += 1
            k += 1

if __name__ == '__main__':
    arr = list(map(int, sys.argv[1:]))
    mergeSort(arr)
    print(arr)
